### PR TITLE
[BugFix] Fix NPE when show mv like

### LINKED ISSUE #20396: SHOW MATERIALIZED VIEWS cause NPE
### Steps to reproduce the behavior (Required)
2023-03-27 02:31:08,396 WARN (starrocks-mysql-nio-pool-16|3336) [StmtExecutor.execute():552] execute Exception, sql SHOW MATERIALIZED VIEWS WHERE NAME LIKE 'mv1';
java.lang.NullPointerException: null
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:430) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:348) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:462) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:728) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_262]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_2

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
3.0

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #20396

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 1]
```diff
@@ -428,10 +428,13 @@ public void execute() throws Exception {
 
                 // sql's blacklist is enabled through enable_sql_blacklist.
                 if (Config.enable_sql_blacklist && !parsedStmt.isExplain()) {
-                    String originSql = parsedStmt.getOrigStmt().originStmt.trim().toLowerCase().replaceAll(" +", " ");
-
-                    // If this sql is in blacklist, show message.
-                    SqlBlackList.verifying(originSql);
+                    OriginStatement origStmt = parsedStmt.getOrigStmt();
+                    if (origStmt != null) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:432]: in which case, origStmt will be null?

**Review by ABingHuang — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 2]
```diff
@@ -574,10 +577,13 @@ public void execute() throws Exception {
             if (parsedStmt instanceof InsertStmt) {
                 // sql's blacklist is enabled through enable_sql_blacklist.
                 if (Config.enable_sql_blacklist) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:579]: Should check if this is a explain statement too.

**Review by ABingHuang — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**